### DISCUSSION TITLE
How can I prevent Jacobian from updating in Newton method ?

### DISCUSSION THREAD

**Discussion by Juns-Cheng** [idaholab/moose#29517] [Q&A Modules: Solid mechanics] [answered]
Hello. I am using the NEWTON method to solve a large deformation mechanics problem. The linear computation is very fast, and the nonlinear iteration is converging. However, for each nonlinear iteration, Jacobian is computed, which costs too much time in my case. So I am wondering if the update can be disabled, like in the modified Newton-Raphson method.

**Comment by GiudGiud** [comment, marked as answer]
Hello

You can:
- set pc reuse parameters (reuse_preconditioner)
OR
- set the solve type to linear
see the executioner documentation page for the parameters

**Reply by Juns-Cheng** [reply]
I believe that is what I am looking for. But it does not take effect. 
In input file I set:
```
 #######################
[Preconditioning]
  [smp]
    type = SMP
    full = true
    petsc_options_iname = '-pc_type'
    petsc_options_value = 'lu'
  []
[]

[Executioner]
  type = Transient
  solve_type = NEWTON
  petsc_options_iname = '-pc_type'
  petsc_options_value = 'lu'
  end_time = 1
  dt = 0.1
  reuse_preconditioner = true
  reuse_preconditioner_max_linear_its = 20
[]
###############
And the output in console looks like, 

1 Nonlinear |R| = 7.962050e-06

    Computing Jacobian...                                                                [ 22.35 s] [ 2548 MB]
      0 Linear |R| = 7.962050e-06
      1 Linear |R| = 5.312515e-06
      2 Linear |R| = 1.173135e-10
      3 Linear |R| = 4.087898e-12
 2 Nonlinear |R| = 5.870031e-06
    Computing Jacobian...                                                                [ 22.25 s] [ 2560 MB]
      0 Linear |R| = 5.870031e-06
      1 Linear |R| = 5.614287e-06
      2 Linear |R| = 4.428558e-10
      3 Linear |R| = 2.125982e-11
 3 Nonlinear |R| = 3.758159e-06
    Computing Jacobian...                                                                [ 22.45 s] [ 2560 MB]
      0 Linear |R| = 3.758159e-06
      1 Linear |R| = 3.670242e-06
      2 Linear |R| = 6.633646e-10
      3 Linear |R| = 3.735210e-11
```
The Jacobian is still updated.

**Reply by GiudGiud** [reply]
what about the linear solve_type?

we must be missing logic here, reusing the preconditioner but recomputing the Jacobian

**Reply by Juns-Cheng** [reply]
For linear solve_type, Jacobian is computed only once for each sub step, as no nonlinear iteration is involved.

**Reply by GiudGiud** [reply]
That should be the way to do it. 
you could also solve with Newton still but a very tight linear convergence to make sure a second nonlinear one is not needed

**Reply by Juns-Cheng** [reply]
But LINEAR solve_type computes Jacobian only once for each sub step, even if reuse_preconditioner is switched off. And I think LINEAR solve_type is not proper for nonlinear problems.

**Reply by GiudGiud** [reply]
if you problem is nonlinear then you need to recompute the Jacobian on every nonlinear iteration.
I thoight your problem was linear from that previous post, but maybe you moved on to another one?

**Reply by Juns-Cheng** [reply]
I am learning moose by trying different mechanics problems. I just found reuse_preconditioner does work for some models. I changed nothing but the mesh file, and then switching off reuse_preconditioner will cost about twice the time than swithing it on. 
Thank you.
